test/test_framework.py

```diff
@@ -557,52 +557,143 @@ def test_mutable_types(self):
         class SomeObject(Object):
             state = StoredState()
 
-        obj = SomeObject(framework, "1")
+        obj = SomeObject(framework, '1')
         try:
             class CustomObject:
                 pass
             obj.state.foo = CustomObject()
         except AttributeError as e:
-            self.assertEqual(str(e), "attribute 'foo' cannot be set to CustomObject: must be int/dict/list/etc")
+            self.assertEqual(str(e), 'attribute \'foo\' cannot be set to CustomObject: must be int/dict/list/etc')
         else:
-            self.fail("AttributeError not raised")
+            self.fail('AttributeError not raised')
 
         framework.commit()
 
         # Test and validation functions in a list of 2-tuples.
         # Assignment and keywords like del are not supported in lambdas so functions are used instead.
-        test_funcs = [(
-            lambda: setattr(obj.state, 'dict', {}),
-            lambda: self.assertEqual(dict(obj.state.dict), {})
+        test_operations = [(
+            {},                                                            # Operand A.
+            None,                                                          # Operand B.
+            {},                                                            # Expected result.
+            lambda a, b: None,                                             # Operation to perform.
```

Comment:
I didn't realize this would be so far onto the right. In general if things are too far apart we break down to make it reasonable. In this case for instance, having the last line in a different indentation would be better than what we got now, because it become hard to match line content with comment.